landlab/components/bedrock_landslider/bedrock_landslider.py

```diff
@@ -0,0 +1,692 @@
+                (np.sin(np.arctan(ss)) * np.cos(sc_rad))
+                / (1 - np.cos(np.arctan(ss) - sc_rad))
+            )
+            Hc[Hc == 0] = np.nan
+            # Spatial probability
+            p_S = H_el / Hc
+            p_S[np.arctan(ss) <= sc_rad] = 0
+            p_S[p_S > 1] = 1
+
+            # Temporal probability
+            p_t = 1 - np.exp(-dt / self._landslides_return_time)
+
+            # Combined probability
+            p = p_t * p_S
+            slides = np.random.rand(p.size) < p
+
+            # Now, find the critical node, which is the receiver of i_slide
+            # Critical nodes must be unique (a given node can have more receivers...)
+            i_slide = np.unique(
+                self.grid.at_node["flow__receiver_node"][np.where(slides)]
+            )
+            # Remove boundary nodes
+            if not self.landslides_on_boundary_nodes:
+                i_slide = i_slide[(self.grid.node_is_boundary(i_slide) is False)]
+        else:
+            i_slide = np.array(self.critical_sliding_nodes)
+
+        # output variables
+        suspended_Sed = np.float64(0)
+        if self._verbose_landslides:
+            print(f"nbSlides = {len(i_slide)}")
+
+        storeV_cum = 0.0
+        while i_slide.size > 0:
+            ind = 0
+            cP = i_slide[ind]  # Critical Node
+            cP_el = topo[cP]
+
+            # get 8 neighbors and only keep those to active nodes which are upstream
+            nb = np.concatenate(
+                (
+                    self.grid.active_adjacent_nodes_at_node[cP],
+                    self.grid.diagonal_adjacent_nodes_at_node[cP],
+                )
+            )
+            nb = nb[nb != -1]
+            nb_up = nb[topo[nb] > cP_el]
+
+            X_cP = self.grid.node_x[cP]
+            Y_cP = self.grid.node_y[cP]
+
+            distToIni_all = np.sqrt(
+                np.add(
+                    np.square(X_cP - self.grid.node_x[nb_up]),
+                    np.square(Y_cP - self.grid.node_y[nb_up]),
+                )
+            )
+            all_iP_el = topo[nb_up]
+            s_slide_all = (all_iP_el - cP_el) / distToIni_all
+
+            nb_up = nb_up[s_slide_all > self._angle_int_frict]
+            s_slide_all = s_slide_all[s_slide_all > self._angle_int_frict]
+
+            if s_slide_all.size > 0:
+                s_slide = max(s_slide_all)
+                storeV_bed = np.float64(0.0)
+                storeV_sed = np.float64(0.0)
+                upstream = 0
+                uP = nb_up
+                if not self.landslides_on_boundary_nodes:
+                    uP = uP[(self.grid.node_is_boundary(uP) is False)]
+                # Fix sliding angle of particular LS
+                ang_sl = np.float64((self._angle_int_frict + s_slide) / 2)
+                stall = 0
+
+                while uP.size > 0 & (
+                    upstream <= self._max_pixelsize_landslide and stall < 1e4
+                ):
```

Comment:
It's probably better to use a logical *and* here (rather than the bitwise operator, *&*)
```python
while uP.size > 0 and upstream <= self._max_pixelsize_landslide and stall < 1e4):
```

Reply:
I see. Done